akka-cluster-tools/src/multi-jvm/scala/akka/cluster/client/ClusterClientSpec.scala

```diff
@@ -185,6 +263,52 @@ class ClusterClientSpec extends MultiNodeSpec(ClusterClientSpec) with STMultiNod
       enterBarrier("after-3")
     }
 
+    "report events" in within(15 seconds) {
+      runOn(client) {
+        implicit val timeout = Timeout(1.second.dilated)
+        val c = Await.result(system.actorSelection("/user/client").resolveOne(), timeout.duration)
+        val l = system.actorOf(Props(classOf[TestClientListener], c), "reporter-client-listener")
+
+        val expectedContacts = Set(first, second, third, fourth).map(node(_) / "system" / "receptionist")
+        l ! TestClientListener.GetLatestContactPoints
+        fishForMessage() {
+          case LatestContactPoints(contacts) if contacts == expectedContacts ⇒
+            true
+          case _ ⇒
+            l ! TestClientListener.GetLatestContactPoints
+            false
+        }
```

Comment:
Not saying that you should change this, more a FYI, another way to do this is (that we use here and there for this kind of assertion in the cluster tests). Reads more cleanly IMO since the entire block is what is repeated rather than spill over from the previous try:
```
within(some-timeout) {
  awaitAssert {
    val probe = TestProbe()
    l.tell(TestClientListener.GetLatestContactPoints, probe.actorRef)
    probe.expectMsgType[LatestContactPoints].contacts should === (expectedContacts)
  }
}
```